src/uct/ib/base/ib_md.c

```diff
@@ -1485,7 +1485,11 @@ ucs_status_t uct_ib_md_open_common(uct_ib_md_t *md,
     if ((md->config.mr_relaxed_order == UCS_CONFIG_ON) && !IBV_ACCESS_RELAXED_ORDERING) {
         ucs_warn("relaxed order memory access requested but not supported");
     } else if (md->config.mr_relaxed_order == UCS_CONFIG_AUTO) {
-        md->config.mr_relaxed_order = UCS_CONFIG_OFF;
+        if (ucs_cpu_prefer_relaxed_order()) {
+            md->config.mr_relaxed_order = UCS_CONFIG_ON;
```

Comment:
why are we sometimes using md_config and other cases md->config?

Reply:
AFAIU md->config is part of md_config which will be accessible after md open. Isn't it https://github.com/openucx/ucx/commit/21ac89b7008a0e9400d0afcb243ca678c956ef4e?

Reply:
so, we should move mr_relaxed_order from uct_ib_md_ext_config_t to uct_ib_md_config_t, then add "int relaxed_order" to uct_md_t - the difference is it's not tri-state, but bool